<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModbusClientWidget</class>
 <widget class="QWidget" name="ModbusClientWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="grpSettings">
     <property name="title">
      <string>Connection Settings</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Mode:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comboMode">
        <item>
         <property name="text">
          <string>TCP</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>RTU (Serial)</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="stackSettings">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="pageTCP">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLineEdit" name="txtIp">
            <property name="text">
             <string>127.0.0.1</string>
            </property>
            <property name="placeholderText">
             <string>IP Address</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinPort">
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="value">
             <number>502</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="pageRTU">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QComboBox" name="comboPort"/>
          </item>
          <item>
           <widget class="QComboBox" name="comboBaud">
            <property name="editable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnConnect">
        <property name="text">
         <string>Connect</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QGroupBox" name="grpRead">
       <property name="title">
        <string>Read Requests</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QComboBox" name="comboRegType">
          <item>
           <property name="text">
            <string>Coils</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Discrete Inputs</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Holding Registers</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Input Registers</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spinStartAddr">
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="prefix">
           <string>Addr: </string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSpinBox" name="spinCount">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>125</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
          <property name="prefix">
           <string>Count: </string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="btnRead">
          <property name="text">
           <string>Read</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QCheckBox" name="chkAutoRead">
          <property name="text">
           <string>Scan (1s)</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTableWidget" name="tableRegisters">
     <property name="columnCount">
      <number>3</number>
     </property>
     <column>
      <property name="text">
       <string>Address</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Value (Dec)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Value (Hex)</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="lblStatus">
     <property name="text">
      <string>Disconnected</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
